source/ice_api.c

```diff
@@ -0,0 +1,1835 @@
+
+    if( stunRetStatus == STUN_RESULT_OK )
+    {
+        if( pIceAgent->isControlling == 0 )
+        {
+            stunRetStatus = StunSerializer_AddAttributeIceControlled( &( stunCxt ),
+                                                                      pIceAgent->tieBreaker );
+        }
+        else
+        {
+            stunRetStatus = StunSerializer_AddAttributeIceControlling( &( stunCxt ),
+                                                                       pIceAgent->tieBreaker );
+        }
+    }
+
+    if( stunRetStatus == STUN_RESULT_OK )
+    {
+        retStatus = Ice_PackageStunPacket( pIceAgent,
+                                           &( stunCxt ),
+                                           ( uint8_t * ) pIceAgent->localPassword,
+                                           ( uint32_t ) strlen( pIceAgent->localPassword ) * sizeof( char ),
+                                           pSendStunMessageBufferLength );
+    }
+
+    if( retStatus == ICE_RESULT_OK )
+    {
+        *ppSendStunMessageBuffer = &( pIceAgent->stunMessageBuffer[ 0 ] );
+    }
+
+    return retStatus;
+}
+
+/*------------------------------------------------------------------------------------------------------------------*/
+
+
+/*
+ +-----+-----+-----+-----+-----+
+ |     | BIT3| BIT2| BIT1| BIT0|
+ +-----+-----+-----+-----+-----+
+ |
+ |  This depicts the connectivityChecks in a candidate pair, these 4 bits show which bit stands for which STUN request/ response.
+ |
+ |   1. BIT0 - STUN request from local candidate to remote candidate.
+ |   2. BIT1 - STUN response from remote candidate to local candidate.
+ |   3. BIT2 - STUN request from remote candidate to local candidate.
+ |   4. BIT3 - STUN response from local candidate to remote candidate.
+ |
+ */
+
+/* Ice_HandleStunPacket - This API handles the processing of Stun Response. */
+
+IceStunPacketHandleResult_t Ice_HandleStunPacket( IceAgent_t * pIceAgent,
+                                                  uint8_t * pReceivedStunMessageBuffer,
+                                                  uint32_t receivedStunMessageBufferLength,
+                                                  uint8_t ** ppSendTransactionIdBuffer,
+                                                  uint8_t ** ppSendStunMessageBuffer,
+                                                  uint32_t * pSendStunMessageBufferLength,
+                                                  IceIPAddress_t * pLocalCandidateAddress,
+                                                  IceIPAddress_t * pRemoteCandidateAddress,
+                                                  IceCandidatePair_t ** ppIceCandidatePair )
+{
+    IceResult_t retStatus = ICE_RESULT_OK;
+    IceStunPacketHandleResult_t stunPacketHandleStatus = ICE_RESULT_STUN_DESERIALIZE_OK;
+    StunResult_t stunResult = STUN_RESULT_OK;
+    StunContext_t stunCxt;
+    StunHeader_t stunHeader;
+    StunAttribute_t stunAttribute;
+    IceCandidate_t * pLocalCandidate;
```

Comment:
Can we also align the integer declarations first followed by pointer declarations, this helps in better readability.

Reply:
updated.